### PR TITLE
LooPyGen v1.0.1

### PR DESCRIPTION
**Pull request by sk33z3r**
LooPyGen CLI and Hotfix Update v1.0.1

* Headless CLI container images for dApp automation
* Sanitize error outputs of all sensitive information
* Fix an issue where large collections would time out
* Fix metadata issue when thumbnails are not set
* Add dummy description to generated metadata for collections done outside of the generator
* Better loading screen texts

Full hotfix list: https://github.com/sk33z3r/LooPyGen/issues?q=is%3Aissue+milestone%3A1.0.1

### CODE REVIEW

File: .github/workflows/image-push-main.yml [hunk 1]
```diff
@@ -21,8 +21,15 @@ jobs:
...
           password: ${{ secrets.HUB_KEY }}
       -
-        name: Build and push
+        name: Build and push GUI images
         uses: docker/build-push-action@v2
         with:
           push: true
           tags: sk33z3r/loopygen:${{ env.HUB_TAG }}, sk33z3r/loopygen:latest
+      -
+        name: Build and push GUI images
```
Comment (Montspy) [inline .github/workflows/image-push-main.yml:30]: Build and push _CLI_ images?

File: README.md [hunk 2]
```diff
@@ -6,15 +6,159 @@
...
-1. Python to generate images and metadata according to the standards for the new Layer 2 marketplace.
-2. NodeJS to pre-calculate CIDs on images and metadata files.
-3. Python to run the batch minting and transfer processes.
-4. PHP to tie everything together.
+⚠️ **NEVER SHARE YOUR PRIVATE KEYS OR PASSPHRASES WITH ANYONE** ⚠️
 
 Checkout the [Wiki pages](https://loopygen.ezl2.app) for up to date info on using the app!
 
 Checkout the [issues section](https://github.com/sk33z3r/LooPyGen/issues) to report any problems, make feature requests, or keep up with our progress!
+
```
Comment (Montspy) [inline README.md:14]: Add a short _LooPyGen UI_ section that points to the wiki setup page. Also a message similar to "LooPyGen UI is what most creators will want to use, follow this link for the guides"

File: python/enc-config.py [hunk 3]
```diff
@@ -36,18 +38,56 @@ def main():
...
+            address = input("From Address: ")
+            l1_pk = getpass("Layer 1 Private Key: ")
+            l2_pk = getpass("Layer 2 Private Key: ")
+            phrase = b64encode(getpass("Passphrase: ").encode("utf-8"))
+            token = input("Fee Token [0 = ETH, 1 = LRC]: ")
+
+            config_json = {
+                "sender": address,
+                "private_key": l2_pk,
+                "private_key_mm": l1_pk,
```
Comment (Montspy) [inline python/enc-config.py:81]: python expects the L1 pk to be prefixed with `0x` but MM does not prefix its pk.
We should add the `0x` prefix if it's missing

@Montspy can add some validity checks on the inputs

File: python/enc-config.py [hunk 4]
```diff
@@ -36,18 +38,56 @@ def main():
...
+                "fee_token": token
```
Comment (Montspy) [inline python/enc-config.py:82]: @Montspy will review to make sure the value types are correct (e.g. not strings where python expects ints)

File: python/prepare.py [hunk 5]
```diff
@@ -18,6 +16,7 @@ def parse_args():
     input_grp = parser.add_mutually_exclusive_group(required=True)
     input_grp.add_argument('--file', help='Specify an input file', type=str)
     input_grp.add_argument('--idir', help='Specify an input directory', type=str)
+    parser.add_argument('--royalty_percentage', metavar='PERCENTAGE', help='Specify the royalty percentage, required with --metadata', type=int)
```
Comment (Montspy) [inline python/prepare.py:19]: @Montspy needs to update the wiki with this new argument

**Review by Montspy — CHANGES_REQUESTED** [review]

File: python/prepare.py (hunk 5 above)
Comment (Montspy) [reply python/prepare.py:19]: Done

File: .github/workflows/image-push-main.yml (hunk 1 above)
Comment (sk33z3r) [reply .github/workflows/image-push-main.yml:30]: yar, copy/paste error

File: README.md (hunk 2 above)
Comment (sk33z3r) [reply README.md:18]: Added and tweaked

File: python/enc-config.py (hunk 3 above)
Comment (Montspy) [reply python/enc-config.py:114]: * [x] The prefixing is already handled by python in `transfer.py`
* [x] Validity checks in [4912520](https://github.com/sk33z3r/LooPyGen/pull/87/commits/491252082c91a1c59f9513cbc76741f219c2f205)

File: python/enc-config.py (hunk 4 above)
Comment (Montspy) [reply python/enc-config.py:115]: Type conversions added in [4912520](https://github.com/sk33z3r/LooPyGen/pull/87/commits/491252082c91a1c59f9513cbc76741f219c2f205)

**Review by Montspy — APPROVED** [review]

**Pull request merged**